src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/BaseSalesforceParser.java

```diff
@@ -84,4 +98,252 @@ private String getEmail(Entity entity) {
+    }
+
+    return presentationFormat("Stage: %s", stageName);
+  }
+
+  /**
+   * Return the Close Date from Salesforce json
+   * @param node type JsonNode
+   * @return The Close Date if it exists formatted, null otherwise.
+   */
+  protected SafeString getCloseDateFormatted(JsonNode node) {
+    String closeDate = node.path("CloseDate").asText();
+
+    if (StringUtils.isEmpty(closeDate)) {
+      return SafeString.EMPTY_SAFE_STRING;
+    }
+
+    SimpleDateFormat formatter = new SimpleDateFormat(TIMESTAMP_FORMAT);
+    String closeDateFormat;
+    try {
+      closeDateFormat = formatter.format(formatter.parse(closeDate));
+    } catch (ParseException e) {
+      return SafeString.EMPTY_SAFE_STRING;
+    }
+
+    return presentationFormat("Close Date: %s", closeDateFormat);
+  }
+
+  /**
+   * Return the Account Name from Salesforce json
+   * @param node type JsonNode
+   * @return The Account Name if it exists formatted, null otherwise.
+   */
+  protected SafeString getAccountNameFormatted(JsonNode node) {
+    String accountName = node.path("Account").path("Name").asText();
+
+    if (StringUtils.isEmpty(accountName)) {
+      return SafeString.EMPTY_SAFE_STRING;
+    }
+
+    return presentationFormat("Account Name: %s", accountName);
+  }
+
+  /**
+   * Return the URL from Account json formated.
+   * @param node type JsonNode
+   * @return (<a href="https://symdev1-dev-ed.my.salesforce.com/00146000004oPCcAAM"/>)
+   */
+  protected SafeString getAccountLinkedFormatted(JsonNode node) {
+    String accountLink = node.path("Account").path("Link").asText();
+
+    if (accountLink.isEmpty()) {
+      return SafeString.EMPTY_SAFE_STRING;
+    }
+
+    SafeString finalUrl = presentationFormat(MessageMLFormatConstants.MESSAGEML_LINK_HREF_FORMAT, accountLink.toString());
+
+    return presentationFormat(LINKED_FORMATTED_TEXT, finalUrl);
+  }
+
+  /**
+   * Return the Amount from Salesforce json
+   * @param node type JsonNode
+   * @return The Amount if it exists formatted, null otherwise.
+   */
+  protected SafeString getAmountFormatted(JsonNode node) {
+    String amount = node.path("Amount").asText();
+
+    if (StringUtils.isEmpty(amount)) {
+      return SafeString.EMPTY_SAFE_STRING;
+    }
+
+    return presentationFormat("Amount: %s", amount);
+  }
+
+  /**
+   * Return the Next Step from Salesforce json
+   * @param node type JsonNode
+   * @return The Next Step if it exists formatted, null otherwise.
+   */
+  protected SafeString getNextStepFormatted(JsonNode node) {
+    String nextStep = getOptionalField(node,  "NextStep", "-").trim();
+
+    if (StringUtils.isEmpty(nextStep)) {
+      return SafeString.EMPTY_SAFE_STRING;
+    }
+
+    return presentationFormat("Next Step: %s", nextStep);
+  }
+
+  /**
+   * Return the Probability from Salesforce json
+   * @param node type JsonNode
+   * @return The Probability if it exists formatted, null otherwise.
+   */
+  protected SafeString getProbabilityFormatted(JsonNode node) {
```

Comment:
Make a common method the encapsulates the logic for checking empty, and pass the JSON node and formatting string to it.